datafusion/optimizer/src/optimizer.rs

```diff
@@ -148,6 +149,7 @@ impl Optimizer {
     /// Create a new optimizer using the recommended list of rules
     pub fn new(config: &OptimizerConfig) -> Self {
         let mut rules: Vec<Arc<dyn OptimizerRule + Sync + Send>> = vec![
+            Arc::new(InlineTableScan::new()),
```

Comment:
@alamb @andygrove this might be a good candidate as well to an `Analysis` phase.

Reply:
I agree